logstash-core/src/main/java/org/logstash/ackedqueue/PageFactory.java

```diff
@@ -0,0 +1,80 @@
+package org.logstash.ackedqueue;
+
+import org.logstash.ackedqueue.io.PageIO;
+
+import java.util.BitSet;
+
+public class PageFactory {
+
+    /**
+     * create a new head page object and new page.{@literal {pageNum}} empty valid data file
+     *
+     * @param pageNum the new head page page number
+     * @param queue the {@link Queue} instance
+     * @param pageIO the {@link PageIO} delegate
+     * @return {@link Page} the new head page
+     */
+    public static Page newHeadPage(int pageNum, Queue queue, PageIO pageIO) {
+        return new Page(pageNum, queue, 0, 0, 0, new BitSet(), pageIO, true);
+    }
+
+    /**
+     * create a new head page from an existing {@link Checkpoint} and open page.{@literal {pageNum}} empty valid data file
+     *
+     * @param checkpoint existing head page {@link Checkpoint}
+     * @param queue the {@link Queue} instance
+     * @param pageIO the {@link PageIO} delegate
+     * @return {@link Page} the new head page
+     */
+    public static Page newHeadPage(Checkpoint checkpoint, Queue queue, PageIO pageIO) {
+        Page p = new Page(
```

Comment:
@colinsurprenant can't we simply overload the constructor of `Page` and save the duplication between this method and `newTailPage`?

Reply:
we could be I'd prefer to keep to so that it is obvious in the code that we are creating a new head page instead of creating a page with a `true` value for the `writable` parameter.